OrbitGl/CaptureSerializer.cpp

```diff
@@ -24,94 +29,101 @@
 #include "TimeGraph.h"
 #include "TimerChain.h"
 #include "absl/strings/str_format.h"
+#include "capture_data.pb.h"
+
+using orbit_ui_protos::CallstackEvent;
+using orbit_ui_protos::CallstackInfo;
+using orbit_ui_protos::CaptureInfo;
+using orbit_ui_protos::FunctionInfo;
+using orbit_ui_protos::TimerInfo;
+
+static absl::flat_hash_map<const uint32_t,
+                           std::pair<const std::string, const std::string>>
+    capture_version_to_orbit_versions_map{
```

Comment:
Maintaining this mapping and the ranges feels like quite a burden to me. Honestly I would lean towards Dimitry's proposal of just saving the version string in the protos.

Reply:
It looks a bit confusing for now, because the minimum and maximum version are the same. I think about the situation when we have range of 10 for example captures from different versions that are compatible and trying to open it with a newer version of Orbit. For all 10 we will get the error saying the exact version and to open it we should have all the 10 versions (according to error). But if we show the range, the user knows that it's enough to have just one version for all the captures. But it depends also on how this map looks like: if we always have min = max it doesn't make sense at all.

Reply:
It's not so much about how it looks, I'm more unsure about whether it's useful to maintain such a map in relation to the fact that it has to be manually maintained and updated for every release. Considering that we're still in beta, we're not committing to maintaining compatibility yet, and that we are probably breaking compatibility at every release, maybe for user it's just better to get a message saying "This capture was created with version ... and is not compatible with the current version" (by saving the version string in the proto) than suggesting them a range of versions to use to open it.
But I might very well be wrong here. Does anyone else have any thoughts about this?

Reply:
True, having such a message might be very convenient for the user. So probably we want to introduce this map (or something similar at some point). But for me it feels a bit premature to do it now. We are not loosing anything by postponing the introduction of such a map to the point where we introduce the first (backwards-compatible) breaking change. And I hope at that point we will have a concept on how to handle breaking changes and for how long and so on.

Reply:
Ok, let's postpone it for now. I removed the map and changed int to string for `version` field in "capture_data.proto".